### PR TITLE
Add Sampling Probability Correction layer.

### PR DESCRIPTION
**Pull request by hertschuh**
This layer corrects logits based on sampling probability.

### CODE REVIEW

File: keras_rs/src/layers/retrieval/sampling_probability_correction.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+from keras_rs.src import types
+from keras_rs.src.api_export import keras_rs_export
+
+
+@keras_rs_export("keras_rs.layers.SamplingProbablityCorrection")
+class SamplingProbablityCorrection(keras.layers.Layer):
+    """Sampling probability correction.
+
+    Args:
+        epsilon: Small float added to sampling probability to avoid taking the
```
Comment (abheesht17) [inline keras_rs/src/layers/retrieval/sampling_probability_correction.py:15]: I've been thinking about our doc-strings. In the rest of Keras, the format is `{arg_name}: {dtype}. {description}`. For example, https://github.com/keras-team/keras-hub/blob/master/keras_hub/src/models/f_net/f_net_backbone.py#L41.
I wonder if we should do the same. Or because we are doing static typing, it'll show up in the generated documentation anyway?

File: keras_rs/src/layers/retrieval/sampling_probability_correction.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+            log of zero. Defaults to 1e-3.
+        **kwargs: Args to pass to the base class.
```
Comment (abheesht17) [inline keras_rs/src/layers/retrieval/sampling_probability_correction.py:17]: Is it standard practice in Keras to specify kwargs in the doc-string? Haven't seen it elsewhere.

File: keras_rs/src/layers/retrieval/sampling_probability_correction.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+from keras import ops
+
+from keras_rs.src import types
+from keras_rs.src.api_export import keras_rs_export
+
+
+@keras_rs_export("keras_rs.layers.SamplingProbablityCorrection")
+class SamplingProbablityCorrection(keras.layers.Layer):
+    """Sampling probability correction.
+
```
Comment (abheesht17) [inline keras_rs/src/layers/retrieval/sampling_probability_correction.py:13]: Should we write a more descriptive doc-string?

File: keras_rs/src/layers/retrieval/sampling_probability_correction_test.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+from keras_rs.src import testing
+from keras_rs.src.layers.retrieval import sampling_probability_correction
+
+
+class SamplingProbabilityCorrectionTest(testing.TestCase):
+    def test_call(self):
+        shape = (10, 20)  # (num_queries, num_candidates)
+        rng = keras.random.SeedGenerator(42)
+        logits = keras.random.uniform(shape, seed=rng)
+        probs = keras.random.uniform(shape[1:], seed=rng)
```
Comment (abheesht17) [inline keras_rs/src/layers/retrieval/sampling_probability_correction_test.py:15]: We can move these to `setUp()` and reuse them in other testcases

**Review by abheesht17 — COMMENTED** [review]
Thanks for the PR! Looks good to me. Just some nits

File: keras_rs/src/layers/retrieval/sampling_probability_correction.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+import keras
+from keras import ops
+
+from keras_rs.src import types
+from keras_rs.src.api_export import keras_rs_export
+
+
+@keras_rs_export("keras_rs.layers.SamplingProbablityCorrection")
+class SamplingProbablityCorrection(keras.layers.Layer):
```
Comment (abheesht17) [inline keras_rs/src/layers/retrieval/sampling_probability_correction.py:11]: Oh, also, we need a `get_config()` method

File: keras_rs/src/layers/retrieval/sampling_probability_correction.py (hunk 1 above)
Comment (hertschuh) [reply keras_rs/src/layers/retrieval/sampling_probability_correction.py:15]: Oh, that's a good question. I don't know if the keras.io scripts handle type annotations.

In code completion, you see the arguments with their types and default values. That's good. Except that `types.Tensor` for instance is replaced by `Any`, because that's how it's defined to not depend on all backends. So maybe for that reason too it makes sense to add the types here.

File: keras_rs/src/layers/retrieval/sampling_probability_correction.py (hunk 2 above)
Comment (hertschuh) [reply keras_rs/src/layers/retrieval/sampling_probability_correction.py:17]: Some internal tools flags the docstring as being incorrect if it's not in the args. So I took the habit of always adding it even if it typically does not provide useful information.

File: keras_rs/src/layers/retrieval/sampling_probability_correction.py (hunk 5 above)
Comment (hertschuh) [reply keras_rs/src/layers/retrieval/sampling_probability_correction.py:11]: Oh true, I added `epsilon`, which wasn't in the original class, I need a `get_config` now.

File: keras_rs/src/layers/retrieval/sampling_probability_correction.py (hunk 1 above)
Comment (hertschuh) [reply keras_rs/src/layers/retrieval/sampling_probability_correction.py:15]: Added.

File: keras_rs/src/layers/retrieval/sampling_probability_correction.py (hunk 5 above)
Comment (hertschuh) [inline keras_rs/src/layers/retrieval/sampling_probability_correction.py:11]: I almost propagated the typo that's been there for years...

**Review by abheesht17 — APPROVED** [review]
Thanks!
